ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeCleanup/FantomasSettingsConverterProvider.fs

```diff
@@ -0,0 +1,43 @@
+module JetBrains.ReSharper.Plugins.FSharp.Psi.Features.CodeCleanup.FSharpEditorConfig
+
+open System
+open JetBrains.Application
+open JetBrains.Application.Settings
+open JetBrains.Application.Settings.Calculated.Implementation
+open JetBrains.DataFlow
+open JetBrains.Diagnostics
+open JetBrains.Lifetimes
+open JetBrains.ReSharper.Plugins.FSharp.Services.Formatter
+open JetBrains.ReSharper.Psi.EditorConfig
+
+type FantomasSettingsConverter(fantomasSettingsEntry: SettingsIndexedEntry) =
+    let fSharpPrefix = "fsharp_"
+
+    let run (context: SettingsConvertContext<'a>) _ =
+        let context = context.As<SettingsConvertContext<string>>()
+        Assertion.AssertNotNull(context, "Context expected to be SettingsConvertContext<string>")
+
+        for entry in context.SourceData.FindEntriesByPrefix(fSharpPrefix) do
+            let setting = SettingIndex(fantomasSettingsEntry, entry.Key)
+            context.Target.SetValue(setting, entry.Value)
+
+        true
+
+    interface IEditorConfigConverter with
+        member val Category = EditorConfigConverter.ReSharperCategory with get, set
+        member x.IsPropertySupported(property) = property.StartsWith(fSharpPrefix, StringComparison.Ordinal)
+        member x.Convert(context) = run context null
+        member x.ConvertAndCheck(context) = run context context
+        member x.ReverseConvert _ = true
+
+
+[<ShellComponent>]
+type FantomasSettingsConverterProvider(lifetime: Lifetime, schema: SettingsSchema) =
+    let items = CollectionEvents<IEditorConfigConverter>(lifetime, $"{nameof FantomasSettingsConverterProvider}.Items")
+
+    do
+        let fantomasSettingsEntry = schema.GetIndexedEntry<_, _>(fun (key: FSharpFormatSettingsKey) -> key.FantomasSettings)
```

Comment:
Is the type argument list needed?

Reply:
It may be very expensive to pass F# lambda like this during shell startup. Please check shell components startup time in logs, and if this component takes a lot of time, look at `FcsCheckerService` where a workaround for getting the entry directly by name is used.